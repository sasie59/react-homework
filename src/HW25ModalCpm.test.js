import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW25ModalCpm';

test('should HW25ModalCpm', () => {
  const onClick = jest.fn(
    (index) => ({index})
  );
  const {container, getByTestId } = render(
    <HW
      index={0}
      onClick={onClick}
    />);
  userEvent.click(getByTestId('modal-bg'));
  expect(container).toMatchSnapshot();

  userEvent.click(getByTestId('close'));
  expect(container).toMatchSnapshot();
});
