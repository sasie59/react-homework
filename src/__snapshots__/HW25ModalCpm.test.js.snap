// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW25ModalCpm 1`] = `
<div>
  <div
    class="modal-bg"
    data-testid="modal-bg"
    id="modal-bg"
  >
    <div
      class="modal-content modal-content"
    >
      <br />
      <button
        data-testid="close"
        id="button"
        name="button"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;

exports[`should HW25ModalCpm 2`] = `
<div>
  <div
    class="modal-bg"
    data-testid="modal-bg"
    id="modal-bg"
  >
    <div
      class="modal-content modal-content"
    >
      <br />
      <button
        data-testid="close"
        id="button"
        name="button"
      >
        關閉
      </button>
    </div>
  </div>
</div>
`;
